src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3333,6 +3344,11 @@ ss::future<bool> ntp_archiver::do_upload_local(
     if (!may_begin_uploads()) {
         co_return false;
     }
+
+    if (!config::shard_local_cfg().cloud_storage_enable_segment_uploads()) {
+        co_return false;
+    }
```

Comment:
How is this case distinct from the changes in `upload_until_term_change_legacy` above?

Reply:
This code path is used by the adjacent segment merging.